File: Source/Back-end/src/main/java/org/elis/manutenzione/dto/request/ModificaManutenzioneRequest.java
package org.elis.manutenzione.dto.request;

import lombok.Data;

/**
 * DTO per la richiesta di modifica di una manutenzione.
 * Contiene l'ID della manutenzione, la nuova descrizione e il nuovo nome.
 */
@Data
public class ModificaManutenzioneRequest {

    /**
     * L'ID della manutenzione da modificare.
     */
    private long idManutenzione;

    /**
     * La nuova descrizione della manutenzione.
     */
    private String descrizione;

    /**
     * Il nuovo nome della manutenzione.
     */
    private String nome;

    /**
     * L'immagine allegata alla manutenzione.
     */
    private String immagine;
}

File: Source/Front-end/front-end/src/app/model/request/modifica-manutenzione-request.ts
export class ModificaManutenzioneRequest {
  idManutenzione: number;
  descrizione: string;
  nome: string;

  constructor(idManutenzione: number, descrizione: string, nome: string) {
    this.idManutenzione = idManutenzione;
    this.descrizione = descrizione;
    this.nome = nome;
  }
}
